vspace/src/main/java/edu/asu/diging/vspace/web/staff/DownloadsController.java

```diff
@@ -0,0 +1,99 @@
+        } catch(NumberFormatException e) {
+            logger.error("Invalid page number", e);
+        }        
+        Page<ExhibitionSnapshot> downloadsPage = snapshotManager.getAllExhibitionSnapshots(pageNum);
+        model.addAttribute("downloadsList" , downloadsPage.getContent());
+        model.addAttribute("downloadsCurrentPageNumber", Integer.parseInt(downloadsPagenum));
+        model.addAttribute("downloadsTotalPages", downloadsPage.getTotalPages());
+        model.addAttribute("downloadsCount", downloadsPage.getTotalElements());
+        
+        return "exhibition/downloads/downloadList";
+    }
+
+    @RequestMapping(value = "/staff/exhibit/snapshot", method = RequestMethod.POST) 
+    public ResponseEntity<ExhibitionSnapshot> createExhibitionSnapshot(HttpServletRequest request, HttpServletResponse response,  Model model) {
+        ExhibitionSnapshot exhibitionSnapshot = null;
+        try {      
+            exhibitionSnapshot = snapshotManager.triggerExhibitionSnapshotCreation();
+
+            return  ResponseEntity.ok()
+                    .header(HttpHeaders.CONTENT_TYPE, "application/json")
+                    .body(exhibitionSnapshot);
+        } 
+        catch (IOException | SnapshotCouldNotBeCreatedException | InterruptedException e) {
+            logger.error("Could not download exhibition", e);
+            return new ResponseEntity<ExhibitionSnapshot>(exhibitionSnapshot, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    @RequestMapping(value = "/staff/exhibit/download/{id}", method = RequestMethod.GET) 
+    public ResponseEntity<Resource> downloadExhibitionFolder(@PathVariable("id") String id, @RequestParam("folderName") String exhibitionSnapshotFolderName , HttpServletRequest request)
+            throws ExhibitionSnapshotNotFoundException , IOException, FileStorageException {
+        Resource resource = null;
+
+        try {
+            byte[] byteArrayResource = snapshotManager.getExhibitionSnapshot(id);
+            resource = new ByteArrayResource(byteArrayResource);
+            return  ResponseEntity.ok()
+                    .header(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename="+exhibitionSnapshotFolderName+".zip")
+                    .contentLength(resource.contentLength())
+                    .header(HttpHeaders.CONTENT_TYPE, "application/zip")
+                    .body(resource);
+        } catch (ExhibitionSnapshotNotFoundException |  IOException e) {
+            logger.error("Could not download exhibition", e);
+            return new ResponseEntity<Resource>(resource, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
+
+    
+    @RequestMapping(value = "/staff/exhibit/download/checkStatus/{id}", method = RequestMethod.GET) 
+    public ResponseEntity<Boolean> exhibitionDownloadStatus(@PathVariable("id") String id, @RequestParam("folderName") String exhibitionDownloadFolderName , HttpServletRequest request) throws ExhibitionSnapshotNotFoundException {
```

Comment:
should probably be in the controller with `createExhibitionSnapshot`

Reply:
wait, what's the difference between this endpoint and `getLatestSnapshotStatus`? Would one method returning the status of a task not be enough?